services/121-service/src/registration/services/registrations-pagination.service.ts

```diff
@@ -164,14 +164,28 @@ export class RegistrationsPaginationService {
 
     queryBuilder = this.addPaymentFilter(queryBuilder, query);
 
-    // Replacing the filter on referenceId is needed in specific cases where the referenceId only consists of numbers.
-    // The !query.filter.referenceId.includes('$') is needed to check if the query doesn't contain an operator like '$ilike'.
-    // If the filter has a '$' we don't need to replace the filter
-    if (query?.filter?.referenceId && !query.filter.referenceId.includes('$')) {
-      queryBuilder.andWhere('CAST("referenceId" AS TEXT) = :referenceId', {
-        referenceId: query.filter.referenceId,
-      });
-      delete query.filter.referenceId;
+    // Replacing the filter on referenceId is needed in specific cases where the referenceId only consists of numbers
+    // or a string that javascript could mistakenly interpret as a number (ie. "651581942751358e5")
+    if (
+      query?.filter?.referenceId &&
+      typeof query.filter.referenceId === 'string'
+    ) {
+      if (!query.filter.referenceId.includes('$')) {
+        // The !query.filter.referenceId.includes('$') is needed to check if the query doesn't contain an operator like '$ilike'.
+        // If the filter has a '$' we don't need to replace the filter
+        queryBuilder.andWhere('CAST("referenceId" AS TEXT) = :referenceId', {
+          referenceId: query.filter.referenceId,
+        });
+        delete query.filter.referenceId;
+      } else if (query.filter.referenceId.includes('$eq')) {
```

Comment:
```suggestion
      } else if (query.filter.referenceId.includes('$eq:')) {
```

Should we align this with the `split`-operation?
To prevent 'false positives' to fall between the cracks?

Reply:
lol too late, but I doubt it will be an issue

Reply:
Yeah...
Lets calculate the changes of a randomly generated reference-ID with `$eq` somewhere in it...

Reply:
Well actually... we're talking about "sanitizing user-input", so its not a matter of chance...
But a matter of _risk_.

What would a user be able to trigger when searching with a filter like: `"?referenceId=abc$eq:sudo-get-me-all-the-money();"`... 🤔

Reply:
I'd say leave a comment in the "Evaluate" task with this and your other reflections. I don't wanna invest time into a perfect hack - I'd rather invest time in a better long-term solution.